CMakeLists.txt

```diff
@@ -0,0 +1,112 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(minisketch
+  VERSION 0.0.1
+  DESCRIPTION "A library for BCH-based set reconciliation"
+  HOMEPAGE_URL "http://github.com/sipa/minisketch/"
+  LANGUAGES CXX
+)
+
+if(DEFINED CMAKE_CXX_STANDARD)
+  if(CMAKE_CXX_STANDARD EQUAL 98 OR CMAKE_CXX_STANDARD LESS 11)
+    message(FATAL_ERROR "This project requires at least C++11")
+  endif()
+else()
+  set(CMAKE_CXX_STANDARD 11)
+endif()
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+if(NOT DEFINED CMAKE_CXX_EXTENSIONS)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+
+option(MINISKETCH_TESTS "Build tests." ON)
+option(MINISKETCH_BENCHMARK "Build benchmark." OFF)
+
+set(supported_fields "2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64")
+set(MINISKETCH_FIELDS ${supported_fields} CACHE STRING "Comma-separated list of field sizes to build. Default=all. Available sizes: ${supported_fields}.")
+string(REPLACE "," ";" supported_field_list "${supported_fields}")
+string(REPLACE "," ";" provided_field_list "${MINISKETCH_FIELDS}")
+foreach(field IN LISTS supported_field_list)
+  if(NOT field IN_LIST provided_field_list)
+    set(have_disabled_fields TRUE)
+    add_compile_definitions(DISABLE_FIELD_${field})
+  endif()
+endforeach()
+
+set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
+
+if(MSVC)
+  add_compile_options(/Zc:__cplusplus /W4 /wd4060 /wd4065 /wd4100 /wd4127 /wd4146 /wd4244 /wd4267 /wd4310)
+else()
+  add_compile_options(-Wall)
+endif()
+
+if(MINGW)
+  add_link_options(-static)
```

Comment:
I don't think we should port this functionality. Instead we should use the CMake shared/static selection machinery.

Reply:
This affects only benchmark/test binaries and allows to run them on the build system via Wine.

Reply:
Consider
```
cmake -B build \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_SYSTEM_NAME=Windows \
  -DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++-posix
cmake --build build --target test
wine ./build/src/test.exe
```
or
```
cmake -B build \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_SYSTEM_NAME=Windows \
  -DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++-posix \
  -DBUILD_SHARED_LIBS=YES
cmake --build build --target test
wine ./build/src/test.exe
```